src/components/update-service/update-service.controller.ts

```diff
@@ -0,0 +1,355 @@
+      hidden: _.get(this.ctrl.serviceClass, 'spec.planUpdatable', false) || _.size(this.ctrl.servicePlans) < 2,
+      allowed: true,
+      valid: true,
+      allowClickNav: true,
+      onShow: this.showPlan
+    };
+    this.configStep = {
+      label: 'Configuration',
+      id: 'configure',
+      view: 'update-service/update-service-configure.html',
+      hidden: false,
+      allowed: true,
+      valid: false,
+      allowClickNav: true,
+      onShow: this.showConfig
+    };
+    this.reviewStep = {
+      label: 'Results',
+      id: 'results',
+      view: 'update-service/update-service-results.html',
+      hidden: false,
+      allowed: false,
+      valid: true,
+      prevEnabled: false,
+      allowClickNav: false,
+      onShow: this.showResults
+    };
+
+    this.ctrl.steps = [this.planStep, this.configStep, this.reviewStep];
+
+    this.ctrl.orderedPlans = _.orderBy(this.ctrl.servicePlans, ['spec.externalMetadata.displayName', 'metadata.name']);
+
+    this.configChanged = false;
+    this.ctrl.displayName = this.$filter('serviceInstanceDisplayName')(this.ctrl.serviceInstance, this.ctrl.serviceClass);
+    this.ctrl.serviceName = _.get(this.ctrl.serviceInstance, 'metadata.name');
+
+    if (this.planStep.hidden) {
+      this.ctrl.hideBackButton = true;
+    }
+
+    this.context = {
+      namespace: _.get(this.ctrl.serviceInstance, 'metadata.namespace')
+    };
+
+    this.origParameterData = angular.copy(_.get(this.ctrl.serviceInstance, 'spec.parameters', {}));
+
+    var promises: any = [];
+
+    _.each(_.get(this.ctrl.serviceInstance, 'spec.parametersFrom'), (parametersSource) => {
+      var secretName = _.get(parametersSource, 'secretKeyRef.name');
+      var secret = _.find(this.secrets, function(nextSecret: any) {
+        return _.get(nextSecret, 'metadata.name') === secretName;
+      });
+
+      if (secret) {
+        this.addParametersFromSecret(secret, parametersSource);
+      } else {
+        promises.push(this.DataService.get("secrets", secretName, this.context).then((secret: any) => {
+          this.addParametersFromSecret(secret, parametersSource);
+          this.secrets.push(secret);
+        }));
+      }
+    });
+
+    this.$q.all(promises).then(() => {
+      this.originalPlan = _.find(this.ctrl.orderedPlans, (orderedPlan: any) => {
+        return _.get(orderedPlan, 'metadata.name') === _.get(this.ctrl.serviceInstance, 'spec.clusterServicePlanRef.name');
+      });
+
+      this.selectPlan(this.originalPlan);
+    });
+  };
+
+  public clearValidityWatcher = () => {
+    if (this.validityWatcher) {
+      this.validityWatcher();
+      this.validityWatcher = undefined;
+    }
+    this.ctrl.reviewStep.allowed = false;
+  };
+
+  public showPlan = () => {
+    this.clearValidityWatcher();
+    this.ctrl.configPageShown = false;
+
+    var updateProperties: any = _.get(this.ctrl.parameterSchema, 'properties');
+    this.configStep.hidden = _.size(updateProperties) < 1;
```

Comment:
Just confirming it's not possible to show the dialog without plans or parameter schema. Are we checking for schema `properties` in origin-web-console?

Reply:
We are. Should I add a safe guard here? What should we do? Close the panel? Display a message?

Reply:
I think it's OK, just wanted to confirm.